pkg/innerring/internal/metachain/contracts/meta/method_objects.go

```diff
@@ -0,0 +1,207 @@
+package meta
+
+import (
+	"fmt"
+	"slices"
+
+	"github.com/nspcc-dev/neo-go/pkg/core/interop"
+	"github.com/nspcc-dev/neo-go/pkg/crypto/hash"
+	"github.com/nspcc-dev/neo-go/pkg/smartcontract"
+	"github.com/nspcc-dev/neo-go/pkg/vm/stackitem"
+)
+
+func (m *MetaData) submitObjectPut(ic *interop.Context, args []stackitem.Item) stackitem.Item {
+	const argsNumber = 2
+	if len(args) != argsNumber {
+		panic(fmt.Errorf("unexpected number of args: %d expected, %d given", argsNumber, len(args)))
+	}
+	metaInfoRaw, ok := args[0].Value().([]byte)
+	if !ok {
+		panic(fmt.Errorf("unexpected first argument value: %T expected, %T given", metaInfoRaw, args[0].Value()))
+	}
+	metaInfoSI, err := stackitem.Deserialize(metaInfoRaw)
+	if err != nil {
+		panic(fmt.Errorf("cannot deserialize meta information from byte array: %w", err))
+	}
+	metaInfo, ok := metaInfoSI.Value().([]stackitem.MapElement)
+	if !ok {
+		panic(fmt.Errorf("unexpected deserialized meta information value: expected %T, %T, given", metaInfo, metaInfoSI.Value()))
+	}
+	cID, err := requiredInMap(metaInfo, "cid").TryBytes()
+	if err != nil || len(cID) != smartcontract.Hash256Len {
+		panic("invalid container ID")
+	}
+	if ic.DAO.GetStorageItem(m.ID, append([]byte{metaContainersPrefix}, cID...)) == nil {
+		panic("container does not support chained metadata")
+	}
+
+	sigsVectorsRaw, ok := args[1].Value().([]stackitem.Item)
```

Comment:
This follows current container contract, but eventually it can be removed from parameters, nodes can sign transaction directly.